CMakeLists.txt

```diff
@@ -275,6 +275,10 @@ option (Seastar_TESTING
   "Enable testing targets."
   ON)
 
+include (CMakeDependentOption)
+cmake_dependent_option (Seastar_TESTING_WITH_NETWORKING "Test with networking access" ON
```

Comment:
Is the idea to have this as a "general" option, i.e. try to remove any test calling external (wan) networks if this is OFF?
Otherwise it should maybe be named after the test group (tls)? I am ok with either intent, but I would prefer a more explanatory description like "Enable tests using network WAN access".

Reply:
> Is the idea to have this as a "general" option, i.e. try to remove any test calling external (wan) networks if this is OFF?

yes. as i put in the cover letter of the 2nd revision, i plan to apply this option to DNS tests in a follow-up change.

> Otherwise it should maybe be named after the test group (tls)? I am ok with either intent, but I would prefer a more explanatory description like "Enable tests using network WAN access".

i'd prefer using a more widely used name.  see the manpage of mock

> [--enable-network](https://www.mankier.com/1/mock#--enable-network)
> Enable networking. If you want to have reproducible builds then your builds should run without a network. This option overrides config_opts['rpmbuild_networking'] and config_opts['use_host_resolv'], setting both True.

see https://www.mankier.com/1/mock

and the manpage of pbuilder

> [--use-network [yes|no]](https://manpages.debian.org/testing/pbuilder/pbuilder.8.en.html#use)
> Specify yes when you do not want to disable network access during build. Network is not available on a Debian buildd, so you might want to keep the default of no. Disabling network access currently only works on Linux.

see https://manpages.debian.org/testing/pbuilder/pbuilder.8.en.html

but if neither of these example is convincing enough, how about using "internet"? as i think, "WAN" is used in the context of describing a network of network (LAN), but what we want to describe is the general access to the external network.

Reply:
Yes, "enable tests accessing internet" seems like a very descriptive naming/description base. Maybe ENABLE_INTERNET_ACCESS_TESTS?
But regarding WAN vs. interwebs; The TLS tests obviously accesses internet proper, but the dns/bind tests don't explicitly do this, they just use the configured DNS to resolve internet addresses, which is somewhat different.